src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,11 +32,16 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
+        String[] splitTrimmedAddress = trimmedAddress.split(", ");
+        int addressLength = splitTrimmedAddress.length;
         this.isPrivate = isPrivate;
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        this.block = addressLength > INDEX_BLOCK ? new Block(splitTrimmedAddress[INDEX_BLOCK]) : new Block("");
+        this.street = addressLength > INDEX_STREET ? new Street(splitTrimmedAddress[INDEX_STREET]) : new Street("");
```

Comment:
Although this is a short way to express if else, it is not preferable because it's readability is not as good as if else